### PR TITLE
TC10_verifying subscription in home page

### PR DESCRIPTION
**Pull request by basant-lazem**

### CODE REVIEW

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 1]
```diff
@@ -13,6 +13,11 @@ public class HomePage {
     private final By featuredItems_div = By.cssSelector("div.features_items");
     private final By recommendedItems_div = By.cssSelector("div.recommended_items");
 
+    private final By subscriptionText = By.xpath("//div[@class=\"single-widget\"]//h2");
+    private final By subscriptionEmail = By.xpath("//*[@id=\"susbscribe_email\"]");
+    private final By subscribeButton = By.xpath("//*[@id=\"subscribe\"]/i");
+    private  final By successMsg = By.xpath("");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:19]: We should not have empty value locator

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 2]
```diff
@@ -13,6 +13,11 @@ public class HomePage {
     private final By featuredItems_div = By.cssSelector("div.features_items");
     private final By recommendedItems_div = By.cssSelector("div.recommended_items");
 
+    private final By subscriptionText = By.xpath("//div[@class=\"single-widget\"]//h2");
+    private final By subscriptionEmail = By.xpath("//*[@id=\"susbscribe_email\"]");
+    private final By subscribeButton = By.xpath("//*[@id=\"subscribe\"]/i");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:18]: Please replace the \" to '

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 3]
```diff
@@ -13,6 +13,11 @@ public class HomePage {
     private final By featuredItems_div = By.cssSelector("div.features_items");
     private final By recommendedItems_div = By.cssSelector("div.recommended_items");
 
+    private final By subscriptionText = By.xpath("//div[@class=\"single-widget\"]//h2");
+    private final By subscriptionEmail = By.xpath("//*[@id=\"susbscribe_email\"]");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:17]: We can use the id locator directly

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java (hunk 2 above)
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:18]: It's not recommended to use * in the xpath, please add the tag names here.

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java [hunk 4]
```diff
@@ -32,5 +37,28 @@ public HomePage validateOnVisibilityOfHomePage() {
...
+    public HomePage clickOnSubscribeButton() {
+        driver.element().click(subscribeButton);
+        return this;
+    }
+//
+//    @Step("Validate on Success Message of Subscription Email")
+//    public HomePage ValidateOnSuccessMessageOfSubscriptionEmail() {
+//        driver.verifyThat().element(successMsg).text().isEqualTo("You have been successfully subscribed!");
+//        return this;
+//    }
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/HomePage.java:61]: Please remove any commented code

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
- Please make sure to execute the pipeline and add the execution link here
- Please make sure to format your code
- Please make sure to have empty line between methods
- Please make sure to not have 2 empty lines anywhere

File: src/main/java/com/gizasystems/automationexercise/pages/HomePage.java (hunk 2 above)
Comment (basant-lazem) [reply src/main/java/com/gizasystems/automationexercise/pages/HomePage.java]: It gives an error when do so.

**Comment by MahmoudElSharkawyGS** [comment]
@basant-lazem Please resolve the conflicts and execute the pipeline again.

**Comment by basant-lazem** [comment]
https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/7883741367

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
You have 10 files changed now :D

**Comment by basant-lazem** [comment]
Pipeline :
https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/8005610384

File: src/main/java/com/gizasystems/automationexercise/pages/CheckOutPage.java [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+import com.shaft.driver.SHAFT;
+import io.qameta.allure.Step;
+import org.openqa.selenium.By;
+public class CheckOutPage {
+    private SHAFT.GUI.WebDriver driver;
+
+    // Locators
+    private String url = System.getProperty("baseUrl") + "/checkout";
+    private final By addressVerifiycation_FirstName = By.xpath("//ul[@class = \"address item box\"]//li[@class=\"address_firstname address_lastname\"]");
+    private  final By addressVerifiycation_address1 = By.xpath("/html/body/section/div/div[3]/div/div[1]/ul/li[4]");
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/CheckOutPage.java:12]: Please refactor this xpath to be relative not absolute.

File: src/main/java/com/gizasystems/automationexercise/pages/SignupPage.java [hunk 6]
```diff
@@ -73,8 +73,12 @@ public SignupPage enterAddressInformation(String userAddressOne, String country,
 
     @Step("Click on Continue Button")
     public SignupPage clickOnContinueButton() {
-        driver.element().click(continueBtn_link);
-        GoogleAlert.dismissAlert(driver, continueBtn_link);
+//        driver.browser().refreshCurrentPage();
+//        driver.element().click(continueBtn_link);
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/SignupPage.java:77]: Please remove commented code.

File: src/main/java/com/gizasystems/automationexercise/pages/SignupPage.java [hunk 7]
```diff
@@ -73,8 +73,12 @@ public SignupPage enterAddressInformation(String userAddressOne, String country,
...
+        driver.element().clickUsingJavascript(continueBtn_link);
+
+        new Actions(driver.getDriver()).doubleClick().perform();
+
```
Comment (MahmoudElSharkawyGS) [inline src/main/java/com/gizasystems/automationexercise/pages/SignupPage.java:81]: Why changing the already working code here? Please revert those changes here.

File: src/test/java/com/gizasystems/automationexercise/tests/test/verifyingSubscriptionInHomePage.java [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+    public void beforeClass() {
+        testUser = new SHAFT.TestData.JSON("src/test/resources/testDataFiles/Footer.json");
+    }
+    @BeforeMethod
+    public void beforeMethod() {
+        driver = new SHAFT.GUI.WebDriver();
+        new HomePage(driver)
+                .navigate()
+                .validateOnVisibilityOfHomePage();
+    }
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/test/verifyingSubscriptionInHomePage.java:31]: Configuration methods (Before & After) should be at the bottom of the test class as agreed.

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]
Please make sure to reformat your code.
Please make sure to execute the pipeline and share the link.

**Comment by basant-lazem** [comment]
pipeline :
https://github.com/MahmoudElSharkawyGS/AutomationExercisePractice/actions/runs/8129035917

File: src/test/java/com/gizasystems/automationexercise/tests/test/verifyingSubscriptionInHomePage.java [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+@Story("Verify Subscription in home page")
+
+public class verifyingSubscriptionInHomePage {
+    private SHAFT.GUI.WebDriver driver;
+    private SHAFT.TestData.JSON testUser;
+    private String timeStamp = String.valueOf(System.currentTimeMillis());
+
+    @Test(description = "Verify Subscription is on Home Page")
+    public void VerifySubscriptionInHomePage() {
+        new Footer(driver).verifySubscriptionText(testUser.getTestData("subText")).enterSubscriptionEmail(testUser.getTestData("UserMail.GuiTimeStamp") + timeStamp + "@gizasystems.com").clickOnSubscribeButton().validateOnSuccessMessageOfSubscriptionEmail(testUser.getTestData("successSubscriptionMessage"));
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/test/verifyingSubscriptionInHomePage.java:24]: Please make sure to align the methods chinning in the test case as agreed (a new line for every method for better readability)

**Review by MahmoudElSharkawyGS — COMMENTED** [review]
Don't forget the naming conventions (lowerCamelCase) please.

File: src/test/java/com/gizasystems/automationexercise/tests/verifyingSubscriptionInHomePage.java [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+@Feature("Home page verification")
+@Story("Verify Subscription in home page")
+
+public class verifyingSubscriptionInHomePage {
+
+    private SHAFT.GUI.WebDriver driver;
+    private SHAFT.TestData.JSON testUser;
+    private String timeStamp = String.valueOf(System.currentTimeMillis());
+
+    @Test(description = "Verify Subscription is on Home Page")
```
Comment (MahmoudElSharkawyGS) [inline src/test/java/com/gizasystems/automationexercise/tests/verifyingSubscriptionInHomePage.java:23]: Please add Description and Tms annotations (@TmsLink should have this value ("55512381")) 
image

**Review by MahmoudElSharkawyGS — CHANGES_REQUESTED** [review]

**Pull request merged**